### PR TITLE
Update Azure.Identity with explicit reference and update in all referenced projects

### PR DESCRIPTION
**Pull request by MaximSieburgh**
As mentioned in a previous PR, this is to fix vulnerability https://github.com/advisories/GHSA-5mfx-4wcx-rv27, the Azure.Identity package needs to be upgraded to version >= 1.10.2. 
As I missed a transitive dependency in the last PR, I'm adding an explicit reference to the nuget package instead.
I also updated the package in the Isolated project this time.

### CODE REVIEW

**Comment by MaximSieburgh** [comment]
@scale-tone Could you please approve this one as well? I missed a reference last time :)

**Comment by MaximSieburgh** [comment]
@scale-tone Gentle reminder

File: durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.csproj [hunk 1]
```diff
@@ -9,6 +9,7 @@
 
   </PropertyGroup>
   <ItemGroup>
+    <PackageReference Include="Azure.Identity" Version="1.10.4" />
```
Comment (scale-tone) [inline durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.csproj:12]: Technically, this dependency comes from Microsoft.Azure.WebJobs.Extensions.DurableTask, and it is them who should care.
We're in contact with the dev team about it.
But until then I agree it makes sense to explicitly lock it here.

**Comment by scale-tone** [comment]
Thanks for that, @MaximSieburgh !
Here is your updated beta package: https://www.nuget.org/packages/DurableFunctionsMonitor.DotNetBackend/6.4.0-beta2

**Pull request merged**